src/openforms/tests/test_frontend_utils.py

```diff
@@ -0,0 +1,52 @@
+from django.test import TestCase
+
+from openforms.frontend import get_frontend_redirect_url
+from openforms.submissions.tests.factories import SubmissionFactory
+
+
+class FrontendRedirectTests(TestCase):
+    def test_frontend_redirect_no_hash(self):
+        submission = SubmissionFactory.create(
+            form_url="https://example.com/basepath",
+        )
+
+        redirect_url = get_frontend_redirect_url(
+            submission,
+            "resume",
+            {"action_1": "arg_1", "action_2": "arg_2"},
+        )
+        self.assertEqual(
+            redirect_url,
+            "https://example.com/basepath?_of_action=resume&_of_action_params=action_1%3Darg_1%26action_2%3Darg_2",
+        )
+
+    def test_frontend_redirect_hash(self):
+        submission = SubmissionFactory.create(
+            form_url="https://example.com/basepath#after_hash",
```

Comment:
good to have it (and keep it!), but let's also do the data migration where we clean/sanitize the URLs for existing submissions.

Reply:
Added the migrations, not sure if it covers all cases? Let me know if you think some other cases should be taken into account into this migration